storage/integration_test.go

```diff
@@ -272,7 +272,9 @@ func multiTransportTest(ctx context.Context, t *testing.T,
 	opts ...option.ClientOption) {
 	for transport, client := range initTransportClients(ctx, t, opts...) {
 		t.Run(transport, func(t *testing.T) {
-			defer client.Close()
+			t.Cleanup(func() {
+				client.Close()
```

Comment:
Nice!

Reply:
Yes, turns out this is necessary b/c defer calls run before t.Cleanup calls :)